### ISSUE TITLE
File permissions issue

### ISSUE THREAD

**Issue by vinulohar007** [Azure/azure-storage-fuse#78]
Hi Team,

We are using SUSE Linux Enterprise Server 12. We have mounted the blobstorage on Linux VM. But we are facing issue with file permissions. Problem is described as below,

We have root user and testuser. we have cloned the blobfuse repository and build the blobfuse using .build.sh.

After successfully building blobfuse we have modified the mount.sh file and put uid and gid option with uid  of testuser and testgroup. and then ran ./mount.sh . blob storage is mounted successfully.
When we are login with root account it is showing the proper permissions and when we login with the testuser it is not allowing to write on get into the directory. it is giving permissions denied error.

***With root user**
Server001:~/# ls -ld /blobdbbackup
drwxrwx--- 2 testuser testgroup 4096 Jan  1  1970 /blobdbbackup

***With testuser login *****
Server001 /% ls -lrt
ls: cannot access 'blobdbbackup': Permission denied
total 208
d?????????   ? ?           ?          ?            ? blobdbbackup

Please help us on this.

Thanks,
Vinayak Lohar

**Comment by seguler** [comment]
You will need to also add -o allow_other option in mount.sh to be able to export the filesystem to other users. 

Btw - can you share the commands to compile this on SLES ? Including installing the dependencies. I've had hard time getting it to compile on SLES 12.

**Comment by vinulohar007** [comment]
Hi @seguler,

**Comment by vinulohar007** [comment]
Hi @seguler,

I have already tried by adding the -o allow_other option, it worked with root user. By root user everything is showing correct. But I am having issue when the user (whose permission I have given) is not able to do anything.

for compiling the blobfuse code follow the below steps,

zypper install git cmake3 fuse-devel libcurl-devel gcc gcc-c++ gnutls-devel libgcrypt-devel fuse 
git clone https://github.com/Azure/azure-storage-fuse/
cd /azure-storage-fuse/
./build.sh

After that you need to just modify the connection.cfg file and mount.sh file 


Thanks,
Vinayak

**Comment by seguler** [comment]
So you are mounting with root user, right ?

Can you share the exact options in mount.sh ? We'll try to repro.

Thank you for the steps on SLES!

**Comment by vinulohar007** [comment]
**more mount.sh**
#!/bin/bash
BLOBFS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
cd $BLOBFS_DIR/build
./blobfuse $1 --tmp-path=/mnt/resource/ramdisk/blobfusetmp -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120  **-o uid=1004 -o gid=1001** --config-file=../connection.cfg

**Comment by seguler** [comment]
I don't think we are doing anything with uid and gid you provide. I've just tested it myself anyway; here is mount.sh:
#!/bin/bash
BLOBFS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
cd $BLOBFS_DIR/build
./blobfuse $1 -o allow_other -o uid=1000 -o gid=1000 --use-https=false --tmp-pat
h=/mnt/blobfusetmp/ -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout
=120 --config-file=../connection.cfg

I then mount it via root (sudo)
sudo ./mount /home/seguler/mount

Both of the following works fine, and shows all the permissions...
sudo ls -lart /home/seguler/mount
ls -lart /home/seguler/mount


As far as I understand, -o allow_other should make the mount point available to all the users. Do you have a requirement to mount this as root and access it through testuser ? Why not mount it with testuser instead ?

**Comment by vinulohar007** [comment]
Hi @seguler ,

Thanks for the solution. -o allow_other option worked for me. Now I am able to see the proper permissions and user can do all operations on that directory.

But I am still facing one issue, I am facing issue with the temporary file system. I am running  a live SAP db backup on blob mount. It is of 250GB. I am having temp space of 111G. when backup writes  files of 111G files backup fails with space issue.

Is there any way to write the files directly to blob storage without writing in temporary space, same as it writes in Azure file share.


Thanks,
Vinayak Lohar

**Comment by luisbtejada** [comment]
Hi, I'm trying to give specific users, specific permission and both chmod/chown have no effect. I have mounter using the allow_other option following the Microsoft docs on linux VMs in azure: 

sudo blobfuse ~/mycontainer --tmp-path=/mnt/resource/blobfusetmp  --config-file=/path/to/fuse_connection.cfg -o attr_timeout=240 -o entry_timeout=240 -o allow_other -o negative_timeout=120

Is there a way were with bloblfuse I can give specific users, specific permissions, like similar to what you'd normally do with chmod or chown?

**Comment by vibhansa-msft** [comment]
As of now we do not support chmod and chown, so there is no direct way to change the permissions or ownership on a given blob. We are working on enabling ADLS account access using blobfuse and that will have option to change permission.
